### PR TITLE
Stardew Valley: Add basic text client with UT tab

### PR DESCRIPTION
**Pull request by Jouramie**
## What is this fixing or adding?
This adds a text client specific to Stardew Valley, with integration to Universal Tracker (when the apworld is installed).

This will later be used for future features, like being able to explain why a location or a game item is in logic or not. The idea came to me seeing Figment wondering what was the rule for the Magnet Bait, to be able to fill their Master Baiter Bundle. Exploiting the Stardew rule framework will allow the client to give user-friendly explanation to why an item is in logic or not based on their seed progress.

Note that Tracker Page still lies on a lot of location and is not yet integrated with ER, bundles randomization and season randomization. This will come in a later PR, possibly only after 6.x.x is merged. **Until it is ready, the link in the AP Launcher is removed. It will not be displayed until UT is properly integrated with Stardew, to avoid confusion.**

## How was this tested?
Launched the client with and without Universal Tracker. Connected to a multiworld to make sure allegedly available locations are displayed in the tracker tab. This might seem light, but this PR rely entirely on UT for all the tracking. 

## If this makes graphical changes, please attach screenshots.

image

image

image

### CODE REVIEW

**Comment by ScipioWright** [comment]
This should absolutely wait until support has been put in for ER to work properly with UT, or you are going to greatly increase the number of false reports of UT bugs that the UT devs get.

**Comment by Jouramie** [comment]
@ScipioWright Based on discussion in UT thread, I removed the link from the launcher at least until UT works with
- ER
- Bundles
- Starting season

This PR is still completely valid. The launcher entry is hidden behind a feature flag. I plan to complete the integration in future PR for the sake of doing small incremental changes.

**Review by agilbert1412 — APPROVED** [review]
Sounds like a pretty good first step

File: data/stardew.png
[the remark this answers is not in the document]
Comment (Jouramie) [reply data/stardew.png:1]: Sure I don't mind changing it

[the remark this answers is not in the document]
Comment (agilbert1412) [reply data/stardew.png:1]: Not sure if we should. I was just asking.
Our game is Stardew, so Stardew icon makes sense, but it's also a lot more chaotic than SMAPI has ever been, so the messy mashup sprite also fits.

I really don't know which is the better choice tbh

[the remark this answers is not in the document]
Comment (Jouramie) [reply data/stardew.png:1]: Since the client is hidden from the launcher, that gives us plenty of time to choose the right icon until it's really released

**Pull request closed without merging**